api/sendfild/service/service.go

```diff
@@ -91,6 +97,17 @@ func New(config Config) (*Service, error) {
 }
 
 func (s *Service) SendFil(ctx context.Context, req *pb.SendFilRequest) (*pb.SendFilResponse, error) {
+	isAllowedFromAddr := s.config.AllowEmptyFromAddrs
+	for _, allowedFromAddr := range s.config.AllowedFromAddrs {
+		if allowedFromAddr == req.From {
+			isAllowedFromAddr = true
+			break
+		}
+	}
+	if !isAllowedFromAddr {
+		return nil, status.Error(codes.InvalidArgument, "from address not allowed")
+	}
```

Comment:
Check our allowed list, or accept all from addresses if the config says that is ok.